### PR TITLE
feat: added modal component

### PR DESCRIPTION
**Pull request by srv-pranjal**
# Added Modal component

# Description

Modal

 - SImple modal
 - List modal
 - Confirmation modal
 
image

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+// Responsible for closing the modal by clicking anywhere else on the window
+window.addEventListener("click", (event) => {
+  if (event.target === simpleModal) {
+    simpleModal.style.display = "none";
+  } else if (event.target === listModal) {
+    listModal.style.display = "none";
+  } else if (event.target === confirmModal) {
+    confirmModal.style.display = "none";
+  }
```
Comment (nikhil-sharma-nks) [inline script.js:22]: just a suggestion, you should consider applying onClick listener to all the close button in the modal.

**Review by nikhil-sharma-nks — COMMENTED** [review]
looks good to be merged, nice UI 🔥👍

**Review by tushargoyal812 — COMMENTED** [review]
i think no changes required looks good to merge

File: script.js (hunk 1 above)
Comment (srv-pranjal) [reply script.js:22]: Sure! Will do that when I'll added JS code snippets for components.

**Pull request merged**